contracts/root/predicates/ERC20Predicate.sol

```diff
@@ -0,0 +1,299 @@
+    inputItems = inputItems[3].toList()[logIndex].toList(); // select log based on given logIndex
+
+    // "address" (contract address that emitted the log) field in the receipt
+    address childToken = RLPReader.toAddress(inputItems[0]);
+    bytes memory logData = inputItems[2].toBytes();
+    inputItems = inputItems[1].toList(); // topics
+    // now, inputItems[i] refers to i-th (0-based) topic in the topics array
+    // event Withdraw(address indexed token, address indexed from, uint256 amountOrTokenId, uint256 input1, uint256 output1)
+    require(
+      bytes32(inputItems[0].toUint()) == WITHDRAW_EVENT_SIG,
+      "Not a withdraw event signature"
+    );
+    address rootToken = address(RLPReader.toUint(inputItems[1]));
+    require(
+      msg.sender == address(inputItems[2].toUint()), // from
+      "Withdrawer and burn exit tx do not match"
+    );
+    uint256 exitAmount = BytesLib.toUint(logData, 0); // amountOrTokenId
+    withdrawManager.addExitToQueue(msg.sender, childToken, rootToken, exitAmount, bytes32(0x0), true /* burnt */, age);
+  }
+
+  function startExit(bytes calldata data, bytes calldata exitTx)
+    external
+  {
+    RLPReader.RLPItem[] memory referenceTx = data.toRlpItem().toList();
+    uint256 age = withdrawManager.verifyInclusion(data, 0 /* offset */, false /* verifyTxInclusion */);
+
+    // validate exitTx - This may be an in-flight tx, so inclusion will not be checked
+    ExitTxData memory exitTxData = processExitTx(exitTx);
+
+    // process the receipt of the referenced tx
+    ReferenceTxData memory referenceTxData = processReferenceTx(
+      referenceTx[6].toBytes(), // receipt
+      referenceTx[9].toUint(), // logIndex
+      exitTxData.signer,
+      false /* isChallenge */
+    );
+    require(
+      exitTxData.childToken == referenceTxData.childToken,
+      "Reference and exit tx do not correspond to the same child token"
+    );
+    validateSequential(exitTxData, referenceTxData);
+    age += referenceTxData.age; // @todo use SafeMath
+
+    if (referenceTx.length <= 10) {
+      withdrawManager.addExitToQueue(
+        msg.sender, referenceTxData.childToken, referenceTxData.rootToken,
+        exitTxData.exitAmount, exitTxData.txHash, exitTxData.burnt, age /* priority */);
+      withdrawManager.addInput(age /* exitId or priority */, age /* age of input */, exitTxData.signer);
+      return;
+    }
+
+    // referenceTx.length > 10 means the exitor sent along another input UTXO to the exit tx
+    // This will be used to exit with the pre-existing balance on the chain along with the couterparty signed exit tx
+    uint256 otherReferenceTxAge = withdrawManager.verifyInclusion(data, 10 /* offset */, false /* verifyTxInclusion */);
+    ReferenceTxData memory _referenceTxData = processReferenceTx(
+      referenceTx[16].toBytes(), // receipt
+      referenceTx[19].toUint(), // logIndex
+      msg.sender, // participant
+      false /* isChallenge */
+    );
+    require(
```

Comment:
can we reduce these 3 require statements related to child and rootTokens?

Reply:
Discussed offline.